include/dpp/managed.h

```diff
@@ -72,6 +68,12 @@ namespace dpp {
 		 * @return false objects are the same id
 		 */
 		bool operator!=(const managed& other) const noexcept;
+
+	protected:
+		/**
+		 * @brief Destroy the managed object
+		 */
+		~managed() = default;
```

Comment:
This now breaks the way that derived classes deconstruct now, no? Like, isn't this meant to be polymorphic?

Reply:
Well, by having it as virtual - it creates a vtable which adds 8bytes of ram to every instance of it and its derived classes. But, by not setting it to virtual, if we attempt to delete a derived class via this base classes' destructor - it won't destroy the derived class. By setting this to non-virtual but also protected, we eliminate the risk that someone will attempt to destroy any of these classes using this base classes' destructor.